impl_unix.go

```diff
@@ -301,7 +312,14 @@ func listenUDP(netw, addr string) (c net.Conn, err error) {
 }
 
 // this is close to the connect() function inside stdlib/net
-func connect(fd int, ra syscall.Sockaddr, deadline time.Time) error {
+func connect(ctx context.Context, fd int, ra syscall.Sockaddr, deadline time.Time) error {
+	// NOTE: We currently only use the context to pass in a deadline.
+	// Canceling an epoll_wait call is hard.
```

Comment:
so it doesn't work with a `cancel()` call, right? i'd be wary of this-- then-- because it doesn't really respect ctx?